server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -2911,7 +2911,7 @@ public void restoreFromRepository(Repository repository, ActionListener<Boolean>
      *
      * @return {@code true} if the engine should be flushed
      */
-    boolean shouldPeriodicallyFlush() {
+    public boolean shouldPeriodicallyFlush() {
```

Comment:
why are we making it public ?

Reply:
We could have used other ways to determine the same - for eg Translog reader count.